src/Stratis.Bitcoin.Features.PoA/SlotsManager.cs

```diff
@@ -29,54 +25,79 @@ public class SlotsManager : ISlotsManager
 
         private readonly IFederationManager federationManager;
 
-        private readonly ChainIndexer chainIndexer;
-
-        private readonly ILogger logger;
+        private readonly IFederationHistory federationHistory;
 
-        public SlotsManager(Network network, IFederationManager federationManager, ChainIndexer chainIndexer, ILoggerFactory loggerFactory)
+        public SlotsManager(Network network, IFederationManager federationManager, IFederationHistory federationHistory)
         {
             Guard.NotNull(network, nameof(network));
             this.federationManager = Guard.NotNull(federationManager, nameof(federationManager));
-            this.chainIndexer = chainIndexer;
+            this.federationHistory = Guard.NotNull(federationHistory, nameof(federationHistory));
             this.consensusOptions = (network as PoANetwork).ConsensusOptions;
-            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
         }
 
         /// <inheritdoc />
-        public uint GetMiningTimestamp(uint currentTime)
+        public DateTimeOffset GetMiningTimestamp(ChainedHeader tip, DateTimeOffset timeNow)
         {
-            if (!this.federationManager.IsFederationMember)
```

Comment:
It would be redundant. The check is implicitly done in the next lines of code by virtue of `CurrentFederationKey` that would be `null`:
```
int myIndex = federationMembers.FindIndex(m => m.PubKey == this.federationManager.CurrentFederationKey?.PubKey);
if (myIndex < 0)
    throw new NotAFederationMemberException();
```